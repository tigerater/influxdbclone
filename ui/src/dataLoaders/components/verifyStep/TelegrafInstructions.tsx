// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Decorator
import {ErrorHandling} from 'src/shared/decorators/errors'

// Components
import CodeSnippet from 'src/shared/components/CodeSnippet'

export interface Props {
  token: string
  configID: string
}

@ErrorHandling
class TelegrafInstructions extends PureComponent<Props> {
  public render() {
    const {token, configID} = this.props
    const exportToken = `export INFLUX_TOKEN=${token || ''}`
    const configScript = `telegraf --config ${
      this.origin
    }/api/v2/telegrafs/${configID || ''}`
    return (
      <div className="wizard-step--body" data-testid="setup-instructions">
        <h6>1. Install the Latest Telegraf</h6>
        <p>
          You can install the latest Telegraf by visiting the{' '}
          <a
            href="https://portal.influxdata.com/downloads/"
            target="_blank"
            rel="noopener noreferrer"
          >
            InfluxData Downloads&nbsp;
          </a>
          page . If you already have Telegraf installed on your system, make
          sure it's up to date. You will need version 1.9.2 or higher.
        </p>
        <h6>2. Configure your API Token</h6>
        <p>
          Your API token is required for pushing data into InfluxDB. You can
          copy the following command to your terminal window to set an
          environment variable with your token.
        </p>
        <CodeSnippet copyText={exportToken} label="CLI" />
        <h6>3. Start Telegraf</h6>
        <p>
          Finally, you can run the following command to start the Telegraf agent
          running on your machine.
        </p>
        <CodeSnippet copyText={configScript} label="CLI" />
      </div>
    )
  }

  private get origin(): string {
    return window.location.origin
  }
}

export default TelegrafInstructions
