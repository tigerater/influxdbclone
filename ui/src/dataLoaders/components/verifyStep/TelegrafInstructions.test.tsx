import React from 'react'
import {shallow} from 'enzyme'

import TelegrafInstructions from 'src/dataLoaders/components/verifyStep/TelegrafInstructions'

let wrapper

const setup = (override = {}) => {
  const props = {
    token: '',
    configID: '',
    ...override,
  }

  return shallow(<TelegrafInstructions {...props} />)
}

describe('TelegrafInstructions', () => {
  it('renders', async () => {
    const wrapper = await setup()
    expect(wrapper.exists()).toBe(true)
  })

  it('matches snapshot', () => {
    wrapper = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
