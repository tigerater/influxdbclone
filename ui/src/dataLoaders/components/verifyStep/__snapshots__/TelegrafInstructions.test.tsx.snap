// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegrafInstructions matches snapshot 1`] = `
<div
  className="wizard-step--body"
>
  <h6>
    1. Install the Latest Telegraf
  </h6>
  <p>
    You can install the latest Telegraf by visiting the
     
    <a
      href="https://portal.influxdata.com/downloads/"
      rel="noopener noreferrer"
      target="_blank"
    >
      InfluxData DownloadsÂ 
    </a>
    page . If you already have Telegraf installed on your system, make sure it's up to date. You will need version 1.9.2 or higher.
  </p>
  <h6>
    2. Configure your API Token
  </h6>
  <p>
    Your API token is required for pushing data into InfluxDB. You can copy the following command to your terminal window to set an environment variable with your token.
  </p>
  <CodeSnippet
    copyText="export INFLUX_TOKEN="
    label="CLI"
  />
  <h6>
    3. Start Telegraf
  </h6>
  <p>
    Finally, you can run the following command to start the Telegraf agent running on your machine.
  </p>
  <CodeSnippet
    copyText="telegraf --config http://localhost/api/v2/telegrafs/"
    label="CLI"
  />
</div>
`;
